breaking/entrypoint.sh

```diff
@@ -23,7 +23,32 @@ fi
 echo "flags: $flags"
 
 if [ -n "$flags" ]; then
-    oasdiff breaking "$base" "$revision" $flags
+    output=$(unbuffer oasdiff breaking "$base" "$revision" $flags)
 else
-    oasdiff breaking "$base" "$revision"
+    output=$(unbuffer oasdiff breaking "$base" "$revision")
 fi
+
+echo "$output"
+
+# GitHub Actions limits output to 1MB
+# We count bytes because unicode has multibyte characters
+size=$(echo "$output" | wc -c)
+if [ "$size" -ge "1000000" ]; then
+    echo "WARN: Diff exceeds the 1MB limit, truncating output..." >&2
+    output=$(echo "$output" | head -c $1000000)
+fi
+
+delimiter=$(cat /proc/sys/kernel/random/uuid | tr -d '-')
+
+# Remove ANSI color codes
+output=$(echo "$output" | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2};?)?)?[mGK]//g")
```

Comment:
Do you want to print the output to the CI (github action) without the ANSI? If yes, I think we need to add a new format to oasdiff. WDYT?

Reply:
I figured it would be nice to preserve the colored output for the GitHub actions log, but markdown output for the step summary would be mangled by the escape characters.

I think oasdiff is doing the right thing to detect a pipe and simplifying its output to plain text. For the GitHub outputs, I'd like to have the indented text without colors, so a custom format could work, or a flag to output to a file.

If we want to keep the colored output for GitHub actions log, then we _could_ run the command twice. While technically inefficient and ugly, oasdiff is fast so it wouldn't cause a significant increase in CI minutes.